### PR TITLE
Navigation us3

### PR DESCRIPTION
**Pull request by LHJE**
closes #75 

***Contributors***
Luke

***What This Does***
Ensures logged in users can see a link to their profile page, to the logout link, and that they do not see the log in or register link when logged in.
It also ensures it shows the logged in user a notifying message signifying who they are logged in as.

Note:
This holds the commit information regarding the commits for the US's we've already worked on.  This branch's new commits start at commit 72fdbea.

### CODE REVIEW

**Comment by leahriffell** [comment]
Thanks for working on this Luke! I just have a question on this git flow and how we could avoid seeing changes that are actually tied to a different milestone reflecting as new changes here. 

Priya did you run into this in last time you used this git flow? If we wait till a milestone's changes are merged to dev, will this avoid this? Looks like we could maybe have a lot of merge conflicts once we do merge `navigation` to `development`.

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:7]: Perfect user. Love it

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:20]: These were old tests?

File: spec/features/navigation_spec.rb
Comment (priyapower) [reply spec/features/navigation_spec.rb:7]: I also like that we are utilizing the before :each as often as we can

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:51]: This test and below meet us3 requirement that a user sees "login" and "profile"

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:67]: This test meets us3 requirements of NOT seeing "login" or "register" anymore

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:7]: Perfect user. Love it

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:11]: I also like that we are utilizing the before :each as often as we can

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:20]: These were old tests?

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:51]: This test meets us3 requirement that a user sees "profile"

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:59]: This test meets us3 requirement that a user sees "logout"

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:67]: This test meets us3 requirements of NOT seeing "login" or "register" anymore

File: spec/features/navigation_spec.rb
Comment (priyapower) [inline spec/features/navigation_spec.rb:76]: This test meets us3 requirements for flash message "logged in as <dynamic name>"

File: app/views/layouts/application.html.erb
Comment (priyapower) [inline app/views/layouts/application.html.erb:17]: Loving this switch statement to ensure our logged in users only see the links that make sense

**Review by priyapower — APPROVED** [review]
Love it! I think it is ready to merge

File: spec/features/navigation_spec.rb
Comment (LHJE) [reply spec/features/navigation_spec.rb:7]: The besssst.

File: spec/features/navigation_spec.rb
Comment (LHJE) [reply spec/features/navigation_spec.rb:20]: Yes!  These ones came with the original build.

**Pull request merged**
